internal/controller/serverset/serverset_test.go

```diff
@@ -548,6 +571,243 @@ func Test_serverSetController_ServerSetObservation(t *testing.T) {
+				kube:                 tt.fields.kube,
+				bootVolumeController: tt.fields.bootVolumeController,
+				nicController:        tt.fields.nicController,
+				serverController:     tt.fields.serverController,
+				log:                  tt.fields.log,
+			}
+
+			got, err := e.Create(tt.args.ctx, tt.args.cr)
+
+			assertions := assert.New(t)
+			fakeBootVolumeCtrl := tt.fields.bootVolumeController.(*fakeKubeBootVolumeControlManager)
+			fakeBootVolumeCtrl.AssertExpectations(t)
+
+			fakeServerCtrl := tt.fields.serverController.(*fakeKubeServerControlManager)
+			fakeServerCtrl.AssertExpectations(t)
+
+			fakeNicCtrl := tt.fields.nicController.(*fakeKubeNicControlManager)
+			fakeNicCtrl.AssertExpectations(t)
+
+			assertions.Equalf(tt.wantErr, err, "Wrong error")
+			assertions.Equalf(tt.want, got, "Wrong response")
+			assertions.Equalf(1, len(tt.args.cr.Status.Conditions), "ServerSet should have one condition")
+			assertCondition(t, xpv1.Creating(), tt.args.cr.Status.Conditions[0], "ServerSet has wrong condition")
+		})
+	}
+}
+
+func fakeBootVolumeCtrlEnsureMethod(timesCalled int) kubeBootVolumeControlManager {
+	bootVolumeCtrl := new(fakeKubeBootVolumeControlManager)
+	if timesCalled > 0 {
+		bootVolumeCtrl.
+			On(ensure, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+			Return(nil).
+			Times(timesCalled)
+	}
+	return bootVolumeCtrl
+
+}
+
+func fakeBootVolumeCtrlEnsureMethodReturnsErr() kubeBootVolumeControlManager {
+	bootVolumeCtrl := new(fakeKubeBootVolumeControlManager)
+	bootVolumeCtrl.
+		On(ensure, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+		Return(errAnErrorWasReceived).
+		Times(1)
+	return bootVolumeCtrl
+}
+
+func fakeServerCtrlEnsureMethod(timesCalled int) kubeServerControlManager {
+	serverCtrl := new(fakeKubeServerControlManager)
+	if timesCalled > 0 {
+		serverCtrl.
+			On(ensure, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+			Return(nil).
+			Times(timesCalled)
+	}
+	return serverCtrl
+}
+
+func fakeServerCtrlEnsureMethodReturnsErr() kubeServerControlManager {
+	serverCtrl := new(fakeKubeServerControlManager)
+	serverCtrl.
+		On(ensure, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+		Return(errAnErrorWasReceived).
+		Times(1)
+	return serverCtrl
+}
+
+func fakeNicCtrlEnsureNICsMethod(timesCalled int) kubeNicControlManager {
+	nicCtrl := new(fakeKubeNicControlManager)
+	if timesCalled > 0 {
+		nicCtrl.
+			On(ensureNICs, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+			Return(nil).
+			Times(noReplicas)
+	}
+	return nicCtrl
+}
+
+func fakeNicCtrlEnsureNICsMethodReturnsErr() kubeNicControlManager {
+	nicCtrl := new(fakeKubeNicControlManager)
+	nicCtrl.
+		On(ensureNICs, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+		Return(errAnErrorWasReceived).
+		Times(1)
+
+	return nicCtrl
+}
+
+func assertCondition(t *testing.T, expected xpv1.Condition, actual xpv1.Condition, msg string) {
```

Comment:
will this be used in multiple tests?

Reply:
Hmm it might not. I've written it like that to make it look somehow like the above assertions.